cogboard-app/src/main/kotlin/com/cognifide/cogboard/ssh/SSHClient.kt

```diff
@@ -0,0 +1,181 @@
+import io.vertx.core.eventbus.MessageConsumer
+import io.vertx.core.json.Json
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import java.io.InputStream
+
+class SSHClient : AbstractVerticle() {
+    private lateinit var session: Session
+    private lateinit var jsch: JSch
+    private lateinit var channel: ChannelExec
+    private lateinit var sshInputStream: InputStream
+    private lateinit var consumer: MessageConsumer<JsonObject>
+    override fun start() {
+        registerSSHCommand()
+        super.start()
+    }
+
+    override fun stop() {
+        consumer.unregister()
+        super.stop()
+    }
+
+    private fun registerSSHCommand() {
+        consumer = vertx.eventBus()
+                .consumer<JsonObject>(CogboardConstants.Event.SSH_COMMAND)
+                .handler { message ->
+                    message.body()?.let {
+                        tryToConnect(it)
+                    }
+                }
+    }
+
+    fun tryToConnect(config: JsonObject) {
+        val eventBusAddress = config.getString(CogboardConstants.Props.EVENT_ADDRESS)
+        try {
+            connect(config)
+        } catch (e: JSchException) {
+            vertx.eventBus().send(eventBusAddress, e)
+        }
+    }
+
+    fun connect(config: JsonObject) {
+        val user = config.getString(CogboardConstants.Props.USER) ?: ""
+        val pass = config.getString(CogboardConstants.Props.PASSWORD) ?: ""
+        val token = config.getString(CogboardConstants.Props.TOKEN) ?: ""
+        val key = config.getString(CogboardConstants.Props.SSH_KEY) ?: ""
+        val host = config.getString(CogboardConstants.Props.SSH_HOST) ?: ""
+
+        val authTypesString = config.getString(CogboardConstants.Props.AUTHENTICATION_TYPES)
+
+        val authTypes = if (authTypesString != null) Json.decodeValue(authTypesString)
+        else JsonArray()
+
+        val authenticationType = getAuthenticationType(authTypes as JsonArray, user, token, pass, key)
+
+        createSSHChannel(authenticationType, host, user, config)
+
+        executeCommandAndSendResult(config)
+    }
+
+    private fun createCommand(config: JsonObject): String {
+        val logLines = config.getString(CogboardConstants.Props.LOG_LINES) ?: "0"
+        val logFilePath = config.getString(CogboardConstants.Props.LOG_FILE_PATH) ?: ""
+
+        return "cat $logFilePath | tail -$logLines"
+    }
+
+    private fun createSSHChannel(
+        authenticationType: AuthenticationType,
+        host: String,
+        user: String,
+        config: JsonObject
+    ) {
+        initJsch(authenticationType, host, user, config)
+        val command = createCommand(config)
+
+        if (session.isConnected) {
+            channel = session.openChannel("exec") as ChannelExec
+
+            channel.setCommand(command)
+            channel.inputStream = null
+            sshInputStream = channel.inputStream
+
+            channel.connect(CogboardConstants.Props.SSH_TIMEOUT)
```

Comment:
to też można wrzucić do oddzielnej metody, żeby opisać bardziej do czego służy ten zbiór instrukcji